provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/builder/util/VulnerabilityUtil.java

```diff
@@ -169,6 +171,37 @@ private static String createRemediationVersionText(RemediatingVersionView remedi
         return stringBuilder.toString();
     }
 
+    private static String createUpgradeGuidanceVersionText(UpgradeGuidanceView upgradeGuidanceView) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(upgradeGuidanceView.getVersionName());
+        UpgradeGuidanceRiskView vulnerabilityRisk = upgradeGuidanceView.getVulnerabilityRisk();
+        if (vulnerabilityRisk.hasVulnerabilities()) {
+            stringBuilder.append(" (Vulnerabilities:");
+            boolean previousVulnerabilites = false;
+            previousVulnerabilites = addVulnerabilityCount(stringBuilder, " Critical: ", previousVulnerabilites, vulnerabilityRisk::hasCriticalVulnerabilities, vulnerabilityRisk::getCritical);
+            previousVulnerabilites = addVulnerabilityCount(stringBuilder, " High: ", previousVulnerabilites, vulnerabilityRisk::hasHighVulnerabilities, vulnerabilityRisk::getHigh);
+            previousVulnerabilites = addVulnerabilityCount(stringBuilder, " Medium: ", previousVulnerabilites, vulnerabilityRisk::hasMediumVulnerabilities, vulnerabilityRisk::getMedium);
+            addVulnerabilityCount(stringBuilder, " Low: ", previousVulnerabilites, vulnerabilityRisk::hasLowVulnerabilities, vulnerabilityRisk::getLow);
```

Comment:
Why not pass the true label (e.g. "Critical") and append the spaces and colon inside of `addVulnerabilityCount`?